src/sentry/api/endpoints/organization_events.py

```diff
@@ -14,6 +14,23 @@
 
 logger = logging.getLogger(__name__)
 
+ALLOWED_REFERRERS = {
+    "api.organization-events-v2",
+    "api.dashboards.tablewidget",
+    "api.dashboards.bignumberwidget",
+    "api.discover.transactions-list",
+    "api.discover.query-table",
+    "api.performance.vitals-cards",
+    "api.performance.landing-table",
+    "api.performance.transaction-summary",
+    "api.performance.status-breakdown",
+    "api.performance.vital-detail",
+    "api.performance.durationpercentilechart",
+    "api.trace-view.span-detail",
+    "api.trace-view.errors-view",
+    "api.trace-view.hover-card",
+}
```

Comment:
@shruthilayaj There's a `api.dashboards.worldmapwidget` referrer that's used by the world map Dashboard widget. See the extended class `class OrganizationEventsGeoEndpoint` at the bottom of this file. I believe the `referrer` should be passed for this endpoint.

Reply:
@dashed The `OrganizationEventsGeoEndpoint` inherits from `OrganizationEventsV2EndpointBase` and not `OrganizationEventsV2Endpoint`- so it shouldn't have to be added to the list here. It still comes though in here:
`12:36:29 server  | 12:36:29 [INFO] sentry.utils.snuba: api.dashboards.worldmapwidget.body: `

Reply:
Ah you're right.

Reply:
Can we still validate the referrer for that endpoint? I want to avoid allowing a user to create arbitrary Snuba transacitons: https://github.com/getsentry/sentry/pull/24858/files#diff-084d1fb1e68307632ee00ae568af3c10195b19880bc8b28c49585c1e1f4d5beeR111

Reply:
For some reason I thought this was a more prevalent problem, but I think these are the only two endpoints that allow the referrer parameter to be set via the request for events